### PR TITLE
shader/validation: Move `break if` expression type validation out of `parse`

### PR DESCRIPTION
**Pull request by ben-clayton**
Use the same pattern by the other statement validation tests.

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [x] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by ben-clayton** [comment]
Also adds testing for abstract types.

File: src/webgpu/shader/validation/statement/break_if.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  'i32',
+  'u32',
+  'f32',
+  'f16',
+  'vec2f',
+  'vec3h',
+  'vec4u',
+  'vec3b',
+  'mat2x3f',
+  'mat4x2h',
```
Comment (dj2) [inline src/webgpu/shader/validation/statement/break_if.spec.ts:20]: This removed a bunch of the extra testing for matrices and vectors, which I think is fine, but it also dropped Struct which I think we possibly want to keep. We may also want to add array, texture, atomic and sampler here as well?

Comment (ben-clayton) [reply src/webgpu/shader/validation/statement/break_if.spec.ts:20]: User-declared and non-constructable types require a bit of work with this pattern. I agree, but I'd like to do this for all the statement tests in this directory, as they all follow the same pattern.

Comment (dj2) [reply src/webgpu/shader/validation/statement/break_if.spec.ts:20]: K, can you please file a bug with the details so we don't lose track?

**Review by dj2 — APPROVED** [review]

File: src/webgpu/shader/validation/statement/break_if.spec.ts (hunk 1 above)
Comment (ben-clayton) [reply src/webgpu/shader/validation/statement/break_if.spec.ts:20]: Done: https://github.com/gpuweb/cts/issues/3685

**Pull request merged**